### PR TITLE
Support incremental upload and ratelimit

### PR DESCRIPTION
**Pull request by kqzh**

### CODE REVIEW

File: internal/utils/incrment.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+	"fmt"
+	"io/ioutil"
+	"path"
+	"strconv"
+	"strings"
+)
+
+const (
+	WalExt          = ".wal"
+	CommitLogIdName = "commitlog.id"
```
Comment (pengweisong) [inline internal/utils/incrment.go:13]: may add a word 'file' to indicate it  file name?

File: internal/utils/incrment.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+func LoadIncrementalFiles(srcDir string, commitLogId int64) ([]string, error) {
+	entries, err := ioutil.ReadDir(srcDir)
+	if err != nil {
+		return nil, err
+	}
+
+	// splitId is the largest id which smaller than commitLogId
+	var splitId int64 = -1
+	// hasLogId check if the src dir have commitlog.id
+	hasLogId := false
```
Comment (pengweisong) [inline internal/utils/incrment.go:25]: hasCommitFile?

File: internal/utils/incrment.go [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+	CommitLogIdName = "commitlog.id"
+)
+
+func LoadIncrementalFiles(srcDir string, commitLogId int64) ([]string, error) {
+	entries, err := ioutil.ReadDir(srcDir)
+	if err != nil {
+		return nil, err
+	}
+
+	// splitId is the largest id which smaller than commitLogId
```
Comment (pengweisong) [inline internal/utils/incrment.go:22]: you mean largest WAL start id?

File: internal/utils/incrment.go [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+	for name, id := range walMap {
+		if id >= splitId {
+			filter = append(filter, name)
+		}
+	}
+
+	return filter, nil
+}
+
+func ParseWalName(name string) (int64, error) {
```
Comment (pengweisong) [inline internal/utils/incrment.go:62]: could make it private

File: internal/utils/incrment.go [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+	}
+
+	// splitId is the largest id which smaller than commitLogId
+	var splitId int64 = -1
+	// hasLogId check if the src dir have commitlog.id
+	hasLogId := false
+	walMap := make(map[string]int64)
+
+	for _, entry := range entries {
+		wid, err := ParseWalName(entry.Name())
```
[the remark this answers is not in the document]
Comment (kqzh) [reply internal/utils/incrment.go:29]: ok

**Pull request closed without merging**